### PR TITLE
Implement maktaba#buffer#GetVisualSelection helper

### LINKED ISSUE #132: Add helper to get current visual selection
Plugins sometimes need to grab highlighted text, e.g. to implement visual-mode mappings that do something with highlighted text, but vim doesn't seem to have any convenient support built in. Maktaba could have a helper like `maktaba#buffer#GetVisualSelection()` that returns the text between the `<` and `>` marks.

### PR DESCRIPTION
**Pull request by dbarnett**
Fixes #132.

### CODE REVIEW

File: vroom/buffer.vroom [hunk 1]
```diff
@@ -9,6 +9,23 @@ But before we dive into things, let's get maktaba installed:
...
+that.
+
+  % Line 1<CR>
+  |Line 2<CR>
+  |Line 3<CR>
+  |Line 4<CR>
+  |Line 5<CR>
+  |Line 6
+
+  > 3gg5|v4gg$
```
Comment (malcolmr) [inline vroom/buffer.vroom]: TIL about the `|` movement command.

File: vroom/buffer.vroom [hunk 2]
```diff
@@ -9,6 +9,23 @@ But before we dive into things, let's get maktaba installed:
...
+  :<C-U>echomsg string(split(maktaba#buffer#GetVisualSelection(), "\n", 1))
+  ~ [' 3', 'Line 4']
+
```
Comment (malcolmr) [inline vroom/buffer.vroom]: Could you add a test using visual line mode (`V`)? That reports 2^31-1 for the `'>` column number, which is weird enough that it seems worth covering (though it looks like it'll work fine).

File: autoload/maktaba/buffer.vim [hunk 3]
```diff
@@ -10,6 +10,29 @@ endfunction
...
 ""
+" Gets the text of the current or last visual selection.
+" Useful for visual mode mappings.
+function! maktaba#buffer#GetVisualSelection() abort
+  let [l:lnum1, l:col1] = getpos("'<")[1:2]
+  let [l:lnum2, l:col2] = getpos("'>")[1:2]
+  " 'selection' is a rarely-used option for overriding whether the last
+  " character is included in the selection. Bizarrely, it always affects the
+  " last character even when selecting from the end backwards.
+  if &selection !=# 'inclusive'
```
Comment (malcolmr) [inline autoload/maktaba/buffer.vim]: I assume we can ignore `'virtualedit'`? (see `:help getpos()`)

File: vroom/buffer.vroom (hunk 2 above)
Comment (dbarnett) [inline vroom/buffer.vroom]: Sure, done.

File: autoload/maktaba/buffer.vim (hunk 3 above)
Comment (dbarnett) [inline autoload/maktaba/buffer.vim]: Oh wow, I'd never heard of that setting. I added a case covering it.

I did discover in the process that this is _not_ the same behavior you get if you yank odd visual selections with `virtualedit=all` active. Yank will pad with spaces if you select part of a tab or past the end of a line. I believe our behavior is more intuitive for use in visual mode mappings and scripts.

**Pull request merged**